src/index.js

```diff
@@ -1,45 +1,60 @@
 const React = require('react')
 const ReactDOM = require('react-dom')
 
-const Container = React.createClass({
-  propTypes: {
-    pages: React.PropTypes.array.isRequired,
-  },
-  getInitialState: function () {
-    return {
-      activePageName: this.getCurrentPageFromUrl(),
+class Container extends React.Component {
+
+  constructor(props) {
+
+    super(props)
+
+    this.state = {
+      activePageName: this.getCurrentPageFromUrl,
+      pages: props.pages
     }
-  },
-  getCurrentPageFromUrl () {
+
+    this.handleHashChange = this.handleHashChange.bind(this)
+
+  }
+
+  static propTypes() {
+    pages: React.PropTypes.array.isRequired
+  }
+
+  getCurrentPageFromUrl() {
     const data = window.location.hash.slice(1).match(/\/page\/(.*)\//)
-    const pageTitle = data[1]
+    const pageTitle = data ? data[1] : 'rawr'
     return pageTitle
-  },
-  componentWillMount: function () {
+  }
+
+  componentWillMount() {
     window.addEventListener('hashchange', this.handleHashChange, true)
-  },
-  componentWillUnmount: function () {
+  }
+
+  componentWillUnmount() {
     window.removeEventListener('hashchange', this.handleHashChange)
-  },
-  handleHashChange: function () {
+  }
+
+  handleHashChange() {
```

Comment:
~~~ diff
-  handleHashChange() {
+  handleHashChange = () => {
~~~

Would fix the bind problem.